server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreLockManagerFactory.java

```diff
@@ -31,9 +32,11 @@
 @PublicApi(since = "2.8.0")
 public class RemoteStoreLockManagerFactory {
     private final Supplier<RepositoriesService> repositoriesService;
+    private final RemoteStoreSettings remoteStoreSettings;
 
-    public RemoteStoreLockManagerFactory(Supplier<RepositoriesService> repositoriesService) {
+    public RemoteStoreLockManagerFactory(Supplier<RepositoriesService> repositoriesService, RemoteStoreSettings remoteStoreSettings) {
```

Comment:
could we have just passed the fixed prefix instead of entire RemoteStoreSettings object?

Reply:
There are 2 settings currently, that will only bloat the ctor args list. The purpose of RemoteStoreSettings class is to keep all settings at single place. RemoteStoreSettings object seems to be a better candidate to me.

Reply:
i saw only `getSegmentsPathFixedPrefix` being used in this class and this Lock Manager Factory seems to be tied to segments, hence suggested that we could have just passed the fixed prefix.